src/highdicom/pr/sop.py

```diff
@@ -0,0 +1,1459 @@
+                        lut_info.append((
+                            voi_lut.LUTDescriptor[1],
+                            voi_lut.LUTDescriptor[2],
+                            getattr(voi_lut, 'LUTExplanation', None),
+                            voi_lut.LUTData
+                        ))
+                    lut_id = tuple(lut_info)
+                    by_lut[lut_id].append(ref_im)
+
+            softcopy_voi_luts = []
+            for (width, center, exp, func), im_list in by_window.items():
+                if len(im_list) == len(referenced_images):
+                    # All datasets included, no need to include the referenced
+                    # images explicitly
+                    refs_to_include = None
+                else:
+                    # Include specific references
+                    refs_to_include = ReferencedImageSequence(im_list)
+
+                softcopy_voi_luts.append(
+                    SoftcopyVOILUT(
+                        window_center=center,
+                        window_width=width,
+                        window_explanation=exp,
+                        voi_lut_function=func,
+                        referenced_images=refs_to_include
+                    )
+                )
+
+            for lut_id, im_list in by_lut.items():
+                if len(im_list) == len(referenced_images):
+                    # All datasets included, no need to include the referenced
+                    # images explicitly
+                    refs_to_include = None
+                else:
+                    # Include specific references
+                    refs_to_include = ReferencedImageSequence(im_list)
+
+                luts = [
+                    LUT(
+                        first_mapped_value=fmv,
+                        lut_data=np.frombuffer(
+                            data,
+                            np.uint8 if ba == 8 else np.uint16
+                        ),
+                        lut_explanation=exp
+                    )
+                    for (fmv, ba, exp, data) in lut_id
+                ]
+                softcopy_voi_luts.append(
+                    SoftcopyVOILUT(
+                        referenced_images=refs_to_include,
+                        voi_luts=luts
+                    )
+                )
+
+            if len(softcopy_voi_luts) > 0:
+                self.SoftcopyVOILUTSequence = softcopy_voi_luts
+
+    def _add_icc_profile(
+        self,
+        icc_profile: Optional[ImageCmsProfile] = None
+    ) -> None:
+        """Add elements of ICC Profile module to the dataset.
+
+        Parameters
+        ----------
+        icc_profile: Union[PIL.ImageCms.ImageCmsProfile, None], optional
+            ICC color profile object to include in the presentation state. If
+            none is provided, a standard RGB ("sRGB") profile will be assumed.
+
+        """
+        if self.SOPClassUID in (
+            ColorSoftcopyPresentationStateStorage,
```

Comment:
In case of a color image, we should probably not make any assumption about the color space, but raise an exception

Reply:
I think for pseudo-color images (grayscale images colored using a Palette Color Lookup Table) it should be fine to assume sRGB, although I could not find any specific statement in the standard about that.

@dclunie do you have any thoughts on this?

Reply:
The standard does not specify a default ICC profile (and nor should it, IMHO) - if the creator elects not to manage color (when allowed by the IOD), then color is unmanaged, and the receiver can do whatever it wants - if the creator wants it interpreted as sRGB, they should say so; if the creator is lazy/unknowing and you want your toolkit to specify sRGB for them by default, that's fine too (not sure you need to bother most users with this by raising an exception). The only problem with sRGB is its limited gamut and for use cases in which that is not a factor, then specifying sRGB is more likely to achieve some consistency of appearance than saying nothing (for those IODs for which saying nothing is permitted). Specifying anything other than sRGB may result in inconsistent appearance if the recipient assumes sRGB without checking/applying whatever profile is actually present.

Reply:
Thanks for the clarification @dclunie.

If the image IOD requires an ICC Profile (e.g., VL Whole Slide Microscopy Image), then we can just copy the ICC Profile from the image and include it in the softcopy presentation state instance. The question is what we should do if the image does not contain an ICC Profile. Both the [Color Softcopy Presentation State IOD](https://dicom.nema.org/medical/dicom/current/output/chtml/part03/sect_A.33.2.3.html#para_9c6369e9-5a08-46c4-923f-0070c8589965) and the [Pseudo-Color Softcopy Presentation State IOD](https://dicom.nema.org/medical/dicom/current/output/chtml/part03/sect_A.33.3.3.html#para_7ac819de-a3f2-4292-9810-787a04783ef6) require the inclusion of the ICC Profile (the module is M and the attribute has type 1).

If I understand you correctly, the ICC Profile module should be conditionally required, depending on whether the image IOD requires it, shouldn't it? Otherwise, we will either have to require the user to provide a profile (most users will probably not know what to do) or use a default profile.

Reply:
No, the creator of the presentation state, for which the presence of an ICC profile is mandatory, has to make a decision and supply one. This is what we intended. Typically a presentation state is created in a situation where the image is being displayed, and it is the context of how the image is being displayed that is being recorded. So the ICC profile is required to allow that state to be replicated to produce a consistent appearance. This may not be directly related to the characteristics of the original camera or other acquisition device. So that is why its presence is mandatory, not conditional. Having said that, ICC profiles probably would have been mandatory for every acquired color image too, if they had been around at the time and there had been an understanding of the color consistency problem and the means to solve it.

Reply:
In 542dd63 I now copy the profile from the referenced images if none is provided by the user, rather than the previous behaviour of using an sRGB default. If the original image has no such profile, an error is raised. We may need to revisit what to do in this situation in the future @hackermd

Reply:
Perfect! Thanks @CPBridge!